mne/preprocessing/xdawn.py

```diff
@@ -244,12 +244,14 @@ def fit(self, epochs, y=None):
         if self.correct_overlap == 'auto':
             self.correct_overlap = _check_overlapp(epochs)
 
+        epochs_data = X.reshape(X.shape[0], info['nchan'], X.shape[1] / 
+                                            info['nchan'])
         # Extract signal covariance
         if self.signal_cov is None:
             if self.correct_overlap:
                 sig_data = _construct_signal_from_epochs(epochs)
```

Comment:
Mmmh, this is getting messy. 

This function attempts to reconstruct the `raw` signals from the epochs so as to compute the channels covariance across time. It output `sig_data`, shape `(n_chans, n_times)`. 

This function is smarter than a strict stacking of the trials as different time points: i.e.

`my_sig_data = np.c[epochs._data]`

because it avoid time duplicates (in case there are overlapping events). 

However, it risks creating weird arrays in case the data was constructed from `EpochsArray`, where the `events` timing may not matter to the user.

I would thus recommend using `sig_data = np.c[epochs._data]` for now; We can always add a PR to handle overlapping events.

cc @dengemann @alexandrebarachant @agramfort